### PR TITLE
Drop `certifi`

### PR DESCRIPTION
**Pull request by jakirkham**
TL;DR: `curl` depends on `openssl`, which pulls in the certs we need from `ca-certificates`. `certifi` is not used for this.

Originally, we had planned to use `certifi` to get certs. However, `certifi` is a Python package and as such puts these in `site-packages`, which is something we not only don't want or need, but it makes it more difficult to properly configure things. Subsequently, we created `ca-certificates`, which simply takes the certs included in `certifi` and moves them to the right location. In the long term, `ca-certificates` may be constructed in different ways, but will always provide these certs. We decided to make this a run time dependency of `openssl`. So, anything using `openssl` will automatically get certs. As we did make the certs a separate package and we did not pin them in `openssl`, updating the certs in `ca-certificates` will get picked up by `conda update --all`.

### CODE REVIEW

**Comment by jakirkham** [comment]
As long as CI passes (should be fine), this is ok to merge whenever. This is just some holdover cruft from staged-recipes.

**Comment by jakirkham** [comment]
Running with this as it is again just cruft that should have already been cleared.
